# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Shape-inl.h
# Commit: ec72f104edbf
# Full Hash: ec72f104edbfe88e22e2ebd778cc7890c21fd2ce
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 9 - Move proto from ObjectGroup to BaseShape. r=jonco
#   
#   The ObjectGroup still stores the proto as well but we don't really use it anymore.
#   This is simpler than removing the group's proto now because that would result in
#   "empty" groups and the group table then doesn't make sense anymore.
# ==============================================================================

diff -r cad616508f73 -r ec72f104edbf js/src/vm/Shape-inl.h
--- a/js/src/vm/Shape-inl.h	Sat Mar 06 01:05:22 2021 +0000
+++ b/js/src/vm/Shape-inl.h	Sat Mar 06 01:05:22 2021 +0000
@@ -29,8 +29,9 @@
   cx_->zone()->setKeepShapeCaches(prev_);
 }
 
-inline StackBaseShape::StackBaseShape(const JSClass* clasp, JS::Realm* realm)
-    : clasp(clasp), realm(realm) {}
+inline StackBaseShape::StackBaseShape(const JSClass* clasp, JS::Realm* realm,
+                                      TaggedProto proto)
+    : clasp(clasp), realm(realm), proto(proto) {}
 
 MOZ_ALWAYS_INLINE Shape* Shape::search(JSContext* cx, jsid id) {
   return search(cx, this, id);